pkg/deploy/registry/deploylog/rest.go

```diff
@@ -180,6 +186,31 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 	}, nil
 }
 
+// waitForExistingDeployment will use the timeout to wait for a deployment to appear.
+func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.ReplicationController, error) {
+	var (
+		target *kapi.ReplicationController
+		err    error
+	)
+
+	condition := func() (bool, error) {
```

Comment:
If the user doesn't specify the version raciness is probably ok.  The
scenario I described is for an end user to be able to log and see
everything - if you say "I want to see logs for the next deployment" then
you're implicitly not setting a hard requirement.

On Wed, Jun 22, 2016 at 11:52 AM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/deploy/registry/deploylog/rest.go
> <https://github.com/openshift/origin/pull/9482#discussion_r68080076>:
>
> > @@ -180,6 +186,31 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
> >  	}, nil
> >  }
> >
> > +// waitForExistingDeployment will use the timeout to wait for a deployment to appear.
> > +func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.ReplicationController, error) {
> > +	var (
> > +		target *kapi.ReplicationController
> > +		err    error
> > +	)
> > +
> > +	condition := func() (bool, error) {
>
> A paused deployment or no triggers don't mean we don't have an already
> running deployment. The command you posted is racy atm because we derive
> the version of the config in the server but we can make it bulletproof if
> we derive it in the client (we already send an individual version
> parameter to enable looking at older deployments).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9482/files/53571d0332a00a9aa0df0515d34a867b65273f42#r68080076>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3GEjXlWeclhO7zVH8XpTPTcBgXuks5qOVoqgaJpZM4I7sgr>
> .
>